connector/openshift/openshift.go

```diff
@@ -85,6 +85,10 @@ func (c *Config) OpenWithHTTPClient(id string, logger log.Logger,
 
 	wellKnownURL := strings.TrimSuffix(c.Issuer, "/") + wellKnownURLPath
 	req, err := http.NewRequest(http.MethodGet, wellKnownURL, nil)
+	if err != nil {
+		cancel()
```

Comment:
TODO: wouldn't defer cancel make more sense here?

Reply:
Yeah, maybe. i will change this.